test-sub.sh

```diff
@@ -0,0 +1,228 @@
+popd
+
+# patchset with whitespace errors
+pushd whitespace
+
+echo "abc" > work
+git add work
+git commit -sm 'add abc to work'
+
+echo "def	" >> work
+git add work
+git commit -sm 'add def to work'
+
+echo "ghi	" >> work
+git add work
+git commit -sm 'add ghi to work'
+
+git format-patch --rfc --cover-letter -v1 -3
+sed -i 's/\*\*\* SUBJECT HERE \*\*\*/Patchset with whitespace errors/g' v1-0000-cover-letter.patch
+sed -i 's/\*\*\* BLURB HERE \*\*\*/Patchset with whitespace errors/g' v1-0000-cover-letter.patch
+
+git send-email --confirm=never --smtp-ssl-cert-path=$WORKDIR/certs/fullchain.pem --to setup@localhost.localdomain *.patch
+popd
+
+# patchset with no cover letter
+pushd nocover
+
+echo "abc" > work
+git add work
+git commit -sm 'add abc to work'
+
+echo "def" >> work
+git add work
+git commit -sm 'add def to work'
+
+git format-patch --rfc -v1 -2
+
+git send-email --confirm=never --smtp-ssl-cert-path=$WORKDIR/certs/fullchain.pem --to setup@localhost.localdomain *.patch
+popd
+
+# patchset with no cover letter and corrupt first patch
+pushd nocover-corrupt
+
+echo "abc" > work
+git add work
+git commit -sm 'add abc to work'
+
+echo "def" >> work
+git add work
+git commit -sm 'add def to work'
+
+echo "ghi" >> work
+git add work
+git commit -sm 'add ghi to work'
+
+git format-patch --rfc -v1 -2
+
+git send-email --confirm=never --smtp-ssl-cert-path=$WORKDIR/certs/fullchain.pem --to setup@localhost.localdomain *.patch
+popd
+
+# end of patchset sending
+popd
+
+sleep 1
+
+# investigate grading repo
+git clone http://localhost:$(get_git_port)/grading.git
+pushd grading
+git fetch --tag
+git tag
+
+
+STATUSES=(
+'patchset applies.'
+'patch 1 failed to apply!'
+'whitespace error patch(es) 2,3?'
+'missing cover letter!'
+'missing cover letter and first patch failed to apply!')
+
+# submitted patchses should have statuses in this order
+# assumption: first ID tag is first patch submitted by this script and IDs increase monotonically
+i=0
+for t in $(git tag); do
+	git show -s --oneline $t | grep -q "${STATUSES[$i]}"
+	i=$((i+1))
+done
+
+popd
+
+echo "wait for initial submission deadline (10 secs)"
+sleep 10
+
+# make final submission
+pushd submissions/bob
+
+# good final patchset
+mkdir good-final
+pushd good-final
+
+echo "abc" > work
+git add work
+git commit -sm 'add abc to work'
+
+echo "def" >> work
+git add work
+git commit -sm 'add def to work'
+
+git format-patch --cover-letter -v2 -2
+sed -i 's/\*\*\* SUBJECT HERE \*\*\*/Good patchset final/g' v2-0000-cover-letter.patch
+sed -i 's/\*\*\* BLURB HERE \*\*\*/Good patchset final/g' v2-0000-cover-letter.patch
+
+git send-email --confirm=never --smtp-ssl-cert-path=$WORKDIR/certs/fullchain.pem --to setup@localhost.localdomain *.patch
+popd
+
+# end final submission
+popd
+
+echo "wait for final submission deadline (10 secs)"
+sleep 10
+
+pushd grading
+git fetch --tag
+
+echo $WORKDIR
```

Comment:
maybe this goes in circumference or wherever we had that other script with users bib bob beb and bab.

Reply:
sure though we should put something that does this sort of thing in the main tests

this is just an initial draft though